cpp/include/cudf/column/column_view.hpp

```diff
@@ -493,4 +473,26 @@ class mutable_column_view : public detail::column_view_base {
  *---------------------------------------------------------------------------**/
 size_type count_descendants(column_view parent);
 
+/**---------------------------------------------------------------------------*
+ * @brief Constrcuts a slice of column_view as per requested offset and size.
+ * The new column_view will start at offset and will have the size requested.
+ * @throws `cudf::logic_error` if `slice_offset` with `slice_size` goes beyond
+ * the size of the `column_view`.
+ * @throws `cudf::logic_error` if `slice_size` < 0.
+ * @throws `cudf::logic_error` if `slice_offset` < 0.
+ *
+ * As the views can have offsets, so actual start would be offset of `input` +
+ * offset requested.
+ *
+ * @param input Immutable view of input column to emulate
+ * @param slice_offset The offset from which the new column_view should start.
+ * @param slice_size The size of the column_view requested from the offset.
+ *
+ * @return unique_ptr<column_view> The unique pointer to the column view
+ * constrcuted as per the offset and size requested.
+ *---------------------------------------------------------------------------**/
+column_view slice(column_view const& input,
+                  size_type slice_offset,
```

Comment:
Can we do this as a `begin/end` index instead? That would be more idiomatic with the rest of libcudf functions and C++ in general.